### PR TITLE
BAH-1017 | Preview and onSave event error handling

### PR DESCRIPTION
**Pull request by AlekhyaYalla**
https://bahmni.atlassian.net/browse/BAH-1017

### CODE REVIEW

File: styles/common/_form.scss [hunk 1]
```diff
@@ -102,3 +102,66 @@
     border-color: $gray400;
   }
 }
+
+.form-preview-popup {
+  &-content {
+    width: 90% !important; /* stylelint-disable-line declaration-no-important */
```
Comment (bsneha90) [inline styles/common/_form.scss:108]: why are we using important?

File: styles/common/_form.scss [hunk 2]
```diff
@@ -102,3 +102,66 @@
...
+    height: 80%;
+    padding: 0 !important; /* stylelint-disable-line declaration-no-important */
```
Comment (bsneha90) [inline styles/common/_form.scss:110]: why are we using important?

File: styles/common/_form.scss [hunk 3]
```diff
@@ -102,3 +102,66 @@
...
+    border-radius: 5px 5px 0;
+    border: 1px solid #666 !important; /* stylelint-disable-line declaration-no-important */
```
Comment (bsneha90) [inline styles/common/_form.scss:112]: why are we using important?

File: styles/common/_popUp.scss [hunk 4]
```diff
@@ -85,3 +85,64 @@
...
+        background: #000;
+        color: #FFF;
+        position: absolute;
+        top: 83%;
+        left: 33%;
+      }
+      .placeholder{
+        pointer-events: none;
+        opacity: 0.5;
+        box-shadow: 0 1px 3px rgba(0, 0, 0, 0.5), 0 1px 2px rgba(0, 0, 0, 0.24) !important; /* stylelint-disable-line declaration-no-important */
```
Comment (bsneha90) [inline styles/common/_popUp.scss:109]: why are we using important?

File: test/form-builder/components/FormDetailContainer.spec.js [hunk 5]
```diff
@@ -182,6 +182,18 @@ describe('FormDetailContainer', () => {
     expect(publishButton).to.have.length(0);
   });
 
+  it('should show preview button before get formData', () => {
```
Comment (bsneha90) [inline test/form-builder/components/FormDetailContainer.spec.js:185]: Should we write a test for preview button will appear after publish?

File: styles/common/_form.scss (hunk 1 above)
Comment (AlekhyaYalla) [reply styles/common/_form.scss:108]: We are overriding  the popup default properties, we have used important as we can't change those default things cant be changed

File: styles/common/_form.scss (hunk 2 above)
Comment (AlekhyaYalla) [reply styles/common/_form.scss:110]: We are overriding the popup default properties, we have used important as we can't change those default things cant be changed

File: styles/common/_form.scss (hunk 3 above)
Comment (AlekhyaYalla) [reply styles/common/_form.scss:112]: We are overriding the popup default properties, we have used important as we can't change those default things cant be changed

File: test/form-builder/components/FormDetailContainer.spec.js (hunk 5 above)
Comment (AlekhyaYalla) [reply test/form-builder/components/FormDetailContainer.spec.js:185]: yes added

File: styles/common/_popUp.scss (hunk 4 above)
Comment (AlekhyaYalla) [reply styles/common/_popUp.scss:109]: removing it, it can be avoided

**Pull request merged**